index.html

```diff
@@ -3303,32 +3311,86 @@ <h3><dfn>Writing or pushing content</dfn></h3>
           </li>
           <li>
             If the type of a |record|'s <a>data</a> is not a
-            {{BufferSource}}, [= exception/throw =] a {{TypeError}}
-            and abort these steps.
+            {{BufferSource}} or {{NDEFMessageInit}}, [= exception/throw =] a
+            {{TypeError}} and abort these steps.
           </li>
           <li>
-            Set |arrayBuffer| to |record|'s <a>data</a>.
+            Set |ndef|'s <a>TNF field</a> to `1`
+            ([=well-known type record=]).
           </li>
           <li>
-            Set |length:unsigned long| to |arrayBuffer|.[[\ArrayBufferByteLength]].
+            Set |ndef|'s <a>TYPE field</a> to |record|'s
+            <a>recordType</a>, representing the <a>local type</a> name.
           </li>
           <li>
-            Set |data:byte sequence| to |arrayBuffer|.[[\ArrayBufferData]].
+            If the type of a |record|'s <a>data</a> is a {{BufferSource}},
+            <ol>
+              <li>
+                Set |arrayBuffer| to |record|'s <a>data</a>.
+              </li>
+              <li>
+                Set |length:unsigned long| to |arrayBuffer|.[[\ArrayBufferByteLength]].
+              </li>
+              <li>
+                Set |data:byte sequence| to |arrayBuffer|.[[\ArrayBufferData]].
+              </li>
+              <li>
+                Set the |ndef|'s <a>PAYLOAD LENGTH field</a> to |length|.
+              </li>
+              <li>
+                If |length| > `0`, set the |ndef|'s <a>PAYLOAD field</a>
+                to |data|.
```

Comment:
Is data initialized to null?

Reply:
No, the algorithms builds it, starting from an empty byte sequence.